dvc/system.py

```diff
@@ -102,19 +102,12 @@ def _reflink_linux(src, dst):
 
         FICLONE = 0x40049409
 
-        s = open(src, "r")
-        d = open(dst, "w+")
-
-        try:
-            ret = fcntl.ioctl(d.fileno(), FICLONE, s.fileno())
-        except IOError:
-            s.close()
-            d.close()
-            os.unlink(dst)
-            raise
-
-        s.close()
-        d.close()
+        with open(src, "r") as s, open(dst, "w+") as d:
+            try:
+                ret = fcntl.ioctl(d.fileno(), FICLONE, s.fileno())
+            except IOError:
+                ret = 255
```

Comment:
ret = 255 looks redundant? 

was `os.unlink` important @efiop ? was there some edge case for that?

Reply:
`ret = 255` triggers `os.unlink()` later.

Reply:
it raises an exception right after this assignment unless I'm missing something?

Reply:
Hmm, that is a bug.

Reply:
Fixed that